onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -0,0 +1,446 @@
+            v.symbolic_value() if isinstance(v, TorchScriptTensor) else v for v in value
+        )
+    if isinstance(value, TorchScriptTensor):
+        return value.symbolic_value()
+    # A normal python value
+    return value
+
+
+@beartype
+def _wrap_torch_value_to_tensor(
+    value: torch.Value
+    | dict[str, torch.Value | str | int | float]
+    | list[torch.Value | str | int | float]
+    | tuple[torch.Value | str | int | float, ...]
+) -> ValidArgumentType | dict[str, ValidArgumentType] | list[ValidArgumentType] | tuple[
+    ValidArgumentType, ...
+]:
+    """Wrap torch.Value to TorchScriptTensor."""
+    if isinstance(value, dict):
+        return {
+            k: TorchScriptTensor(v) if isinstance(v, torch.Value) else v
+            for k, v in value.items()
+        }
+    if isinstance(value, list):
+        return [TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value]
+    if isinstance(value, tuple):
+        return tuple(TorchScriptTensor(v) if isinstance(v, torch.Value) else v for v in value)
+    if isinstance(value, torch.Value):
+        return TorchScriptTensor(value)
+    return value
+
+
+def _unwrap_tensors_to_torch_values(tensors):
+    # TODO(justinchuby): Do we really need this?
+    if isinstance(tensors, Sequence):
+        return [_unwrap_tensor_to_torch_value(output) for output in tensors]
+    return _unwrap_tensor_to_torch_value(tensors)
+
+
+class TorchScriptEvaluator(evaluator.Evaluator):
+    """An onnxscript Evaluator that captures the graph into torchscript."""
+
+    def __init__(self, graph: TorchScriptGraph):
+        self._graph: TorchScriptGraph = graph
+
+    @property
+    def graph(self) -> TorchScriptGraph:
+        return self._graph
+
+    @beartype
+    def eval_function(
+        self,
+        function: onnxscript.OnnxFunction,
+        args: Sequence[ValidArgumentType],
+        kwargs: Sequence[ValidArgumentType],
+    ):
+        # args/kwargs are TorchScriptTensor/python built-in based
+        inputs, attributes = _split_args_kwargs_to_input_attr(function, args, kwargs)
+        return self._graph.add_function(function, inputs, attributes)
+
+    def _eval(self, schema: onnx.defs.OpSchema, inputs, attributes, closure: Any):
+        # TODO(justinchuby): Does it really know what the inputs are?
+
+        del closure  # Unused
+        return self._graph.add_op(schema, inputs, attributes)
+
+
+@beartype
+def _add_attribute_to_torchscrpt_node(
+    node: torch.Node, key: str, value: float | int | str | Sequence[float] | Sequence[int]
+):
+    """Initializes the right attribute based on type of value."""
+    if isinstance(value, float):
+        return node.f_(key, value)
+    if isinstance(value, int):
+        return node.i_(key, value)
+    if isinstance(value, str):
+        return node.s_(key, value)
+    if isinstance(value, torch.Tensor):
+        return node.t_(key, value)
+    if isinstance(value, Sequence):
+        if isinstance(value, float):
```

Comment:
Doesn't look right. May be check either value[0] or all values?

Reply:
Good catch thanks

Reply:
Done